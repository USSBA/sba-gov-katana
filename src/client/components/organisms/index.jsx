/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
