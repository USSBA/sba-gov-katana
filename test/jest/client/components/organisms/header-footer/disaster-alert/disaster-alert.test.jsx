import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

import { DisasterAlert } from 'organisms'
import Main from 'client/components/templates/main'

const alertProps1 = {
  buttonText: 'Learn more',
  description: 'This is an alert',
  link: 'https://www.sba.gov',
  onClose: () => {
    alert('closed')
  },
  visible: true,
  spanishTranslation: {
    buttonText: 'Aprende mÃ¡s',
    description: 'Esto es una alerta',
    link: 'https://www.sba.gov'
  }
}

const alertProps2 = {
  buttonText: 'Click here',
  description: 'This is a message',
  link: 'https://www.code.gov',
  onClose: () => {
    alert('close')
  },
  visible: false
}

describe('Disaster Alert', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<DisasterAlert {...alertProps1} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render with its elements', () => {
    const component = mount(<DisasterAlert {...alertProps1} />)
    expect(component.find('#disaster-alert').exists()).toEqual(true)
    expect(component.find('img').exists()).toEqual(true)
    expect(
      component
        .find('.alertLink')
        .at(0)
        .text()
    ).toEqual('Learn more')
  })

  it('should not render', () => {
    const component = mount(<DisasterAlert {...alertProps2} />)
    expect(component.find('#disaster-alert').exists()).toEqual(false)
    expect(component.find('img').exists()).toEqual(false)
    expect(component.find('.alertLink').exists()).toEqual(false)
  })

  it('should handle click event when pressing on the X image', () => {
    window.alert = jest.fn()
    const component = shallow(<DisasterAlert {...alertProps1} />)
    component.find('img').simulate('click')
    expect(window.alert).toHaveBeenCalledWith('closed')
  })

  it('should match snapshot with Spanish content', () => {
    history.pushState({}, null, '?lang=es')

    const component = renderer.create(<DisasterAlert {...alertProps1} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
