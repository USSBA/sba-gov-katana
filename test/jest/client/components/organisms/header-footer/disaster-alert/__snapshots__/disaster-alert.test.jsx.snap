// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disaster Alert should match snapshot with Spanish content 1`] = `
<div
  className="wrapper"
  id="disaster-alert"
>
  <div
    className="alert"
  >
    <div
      aria-hidden="true"
      className="alertIcon fa fa-exclamation-triangle"
    />
    <div
      aria-label="Esto es una alerta"
      className="disasterDescription"
      tabIndex="1"
    >
      Esto es una alerta
    </div>
    <img
      alt="close"
      aria-label="click to close alert"
      className="alertClose"
      onClick={[Function]}
      onKeyDown={[Function]}
      src="test-file-stub"
      tabIndex="3"
    />
    <a
      aria-label="click to learn more about Esto es una alerta"
      className="alertLink"
      href="https://www.sba.gov"
      tabIndex="2"
    >
      Aprende m√°s
    </a>
  </div>
</div>
`;

exports[`Disaster Alert should render correctly 1`] = `
<div
  className="wrapper"
  id="disaster-alert"
>
  <div
    className="alert"
  >
    <div
      aria-hidden="true"
      className="alertIcon fa fa-exclamation-triangle"
    />
    <div
      aria-label="This is an alert"
      className="disasterDescription"
      tabIndex="1"
    >
      This is an alert
    </div>
    <img
      alt="close"
      aria-label="click to close alert"
      className="alertClose"
      onClick={[Function]}
      onKeyDown={[Function]}
      src="test-file-stub"
      tabIndex="3"
    />
    <a
      aria-label="click to learn more about This is an alert"
      className="alertLink"
      href="https://www.sba.gov"
      tabIndex="2"
    >
      Learn more
    </a>
  </div>
</div>
`;
